pkg/app/zipper/tldcache.go

```diff
@@ -34,12 +45,68 @@ func (app *App) probeTopLevelDomains(ms *PrometheusMetrics) {
 	}
 }
 
+type tldPrefix struct {
+	prefix  string
+	nsCount int
+}
+
+func sortedByNsCount(prefixes []string) []string {
+	countedPrefixes := make([]tldPrefix, len(prefixes))
+	for i, prefix := range prefixes {
+		if prefix == "" {
+			countedPrefixes[i] = tldPrefix{
+				prefix:  prefix,
+				nsCount: 0,
```

Comment:
nit: imho it would be simpler to calculate nsCount first, and have single code path for assignment